source/postprocess/sea_level.cc

```diff
@@ -0,0 +1,584 @@
+                    const double geoid_displacement = geoid.evaluate(vertex); // (check sign of geoid_displacement)
+                    const double topography = this->get_geometry_model().height_above_reference_surface(vertex);
+
+                    Point<dim-1> internal_position;
+                    const std::array<double,dim> spherical_position = aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(vertex);
+                    for (unsigned int d = 1; d < dim; ++d)
+                      internal_position[d-1] = spherical_position[d];
+
+                    const double ice_height = ice_height_lookup->get_data(internal_position,0);
+                    const double topography_init = topography_lookup->get_data(internal_position,0);
+                    int ocean_mask = 0;
+                    if (topography_init > 0.0)
+                      ocean_mask = 1;
+                    else
+                      ocean_mask = 0;
+
+                    if (write_to_file)
+                      {
+                        const std::array<double,dim> scoord = aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(vertex);
+                        Point<dim> scoord_new;
+                        scoord_new[0] = scoord[0];
+                        scoord_new[1] = scoord[1];
+                        scoord_new[2] = scoord[2];
+                        output_file << scoord_new << ' '<< nonuniform_sea_level_change << sea_level_offset << geoid_displacement << topography << ice_height << ocean_mask << std::endl;
+                      }
+                    if (nonuniform_sea_level_change > local_max_height)
+                      local_max_height = nonuniform_sea_level_change;
+                    if (nonuniform_sea_level_change < local_min_height)
+                      local_min_height = nonuniform_sea_level_change;
+                  }
+              }
+
+      // Calculate min/max non uniform sea_level change across all processes.
+      const double max_nonuniform_sea_level_change = Utilities::MPI::max(local_max_height, this->get_mpi_communicator());
+      const double min_nonuniform_sea_level_change = Utilities::MPI::min(local_min_height, this->get_mpi_communicator());
+
+      // Write results to statistics file.
+      statistics.add_value ("Minimum non-uniform sea_level change (m)",
+                            min_nonuniform_sea_level_change);
+      statistics.add_value ("Maximum non-uniform sea_level change (m)",
+                            max_nonuniform_sea_level_change);
+      const char *columns[] = { "Minimum non-uniform sea_level change (m)",
+                                "Maximum non-uniform sea_level change (m)"
+                              };
+      for (unsigned int i=0; i<sizeof(columns)/sizeof(columns[0]); ++i)
+        {
+          statistics.set_precision (columns[i], 8);
+          statistics.set_scientific (columns[i], true);
+        }
```

Comment:
We've got a new style:
```suggestion
      for (const auto &column : columns)
        {
          statistics.set_precision (column, 8);
          statistics.set_scientific (column, true);
        }
```